Fix  #1198 "Context menu with menu key broken"

lukefromdc opened the pull request:
Pass "NULL" for event which is passed ultimately to gtk_menu_popup_at_pointer as an invalid event was being passed when the menu key was used to open the icon views context menu

lukefromdc commented:
I just found I have another piece of this to fix: selecting an item within the icon view with the mouse, then hitting the menu key still gets no menu and still gets 
`(caja:22948): Gtk-CRITICAL **: 14:48:42.830: gtk_menu_popup_at_rect: assertion 'GDK_IS_WINDOW (rect_window)' failed
`
So I need to find that function and pass NULL to eel_pop_up_context_menu in that case as well. We need to NOT pass NULL in the pathbar's case, but pass NULL in all of the icon view cases and I just found another one. Will work on that this evening, out of time at the moment. 

If you all want to merge this now I can do that in a second PR, if this is not merged this evening I will add it to this one. Either way works for me

raveit65 commented:
I would prefer to see all fixes in this PR , for complete testing.
If someone needs it urgent, he can extract the commit from this branch.

lukefromdc commented:
Fine be me, should be able to have it all ready tonight

On 4/3/2019 at 3:02 PM, "raveit65" <notifications@github.com> wrote:
>
>I would prefer to see all fixes in this PR , for complete testing.
>If someone needs it urgent, he can extract the commit from this 
>branch.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/1200#issuecomment-
>479618430

lukefromdc commented:
Fixed the issue with selecting an item in an icon view and popping up the context menu with the menu key, and rebased for latest changes in master. Right click menus for pathbar work fine. Using the menu key for locations in the pathbar  does not, but I'm not sure if it ever did, and problems there forced passing the event in the first place. To make that work would require something more complex than this. We don't get right click menus for toolbar items etc at all. 

At any rate, this should be ready to go, do check for any regressions. Only context menus for icon views and for places should be affected in any way.

raveit65 commented:
Little rewording, i only moved the link to bugreport from commit header to commit body.
I hope you agree.
Now i start testing.....

lukefromdc commented:
fine by me

On 4/8/2019 at 1:12 PM, "raveit65" <notifications@github.com> wrote:
>
>Little rewording, i only moved the link to bugreport from commit 
>header to commit body.
>I hope you agree.
>Now i start testing.....
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/1200#issuecomment-
>480921112

raveit65 commented:
Not related to PR because i see that behaviour with 1.22.0 too.
The context-menu from pathbar-buttons doesn't open with keyboard.
Not sure if this was ever working.

sc0w commented:
@raveit65 

> Not sure if this was ever working.

https://github.com/mate-desktop/caja/issues/1198#issuecomment-479386002

culprit commit: https://github.com/mate-desktop/caja/commit/c0f159de3fc1fdce082acef36044016e3f7b3527
